src/cli/CommandLineIface.h

```diff
@@ -15,16 +15,15 @@
 class CommandLineIface : public QObject {
     Q_OBJECT
 private:
-    ModelManager models_;
-    
+    // Event loop that would wait until translation completes
+    QEventLoop eventLoop_;
+
     // Settings, network and translator:
-    Settings settings_;
     Network network_;
+    Settings settings_;
+    ModelManager models_;
```

Comment:
I'm so confused here. In the `CommandLineIface.cpp` we have:
```cpp
, models_(this, &settings_)
```
Models requires settings, but now we have some (inversed?!?!) order dependency? That sounds very very wrong.

Reply:
Looks pretty normal to me. ModelManager needs settings, but isn't the only one that needs settings. You don't want modelmanager to manage its own separate copy, nor do you want to instantiate modelmanager to get access to settings. The only thing that makes it look weirder is that both modelmanager and settings are member variables.

Reply:
But to be fair it does look out of place in Qt context. I think Qt solves this by having all QSettings instances share the same backend and sync them under the hood. We could piggyback on that synchronisation.

Reply:
From here it makes it seem that `settings_` has a hidden dependencies on model manager and can't be used separately.

Reply:
I don't think I follow… Settings.h has no reference to `ModelManager`, and can function without it.

Is it because we pass in a pointer to `settings_` instead of a reference when constructing `models_`? The only argument for that is that it allows you to do `new ModelManager(...)` and have it take ownership of the Settings instance it uses by calling `setParent()`. Like it has ownership of its own Network object. It seems to be the Qt way. But if it was just std C++, I would have passed it by reference. (If only C++ had explicit lifetimes in its typesystem…)

> We could piggyback on that synchronisation.
I was wrong. QSettings does not emit anything when a value is changed, so there is a reason we need to share one Settings instance throughout the application. `Settings` could be singleton, but I generally don't like those.

Reply:
Sorry, my bad. I misunderstood that models need to get initialised before settings, but yeah it's the other way around of course. Really my bad, sorry.